CIME/SystemTests/mvk.py

```diff
@@ -9,35 +9,129 @@
 import os
 import json
 import logging
-
 from distutils import dir_util
 
-import CIME.test_status
-import CIME.utils
+from CIME import test_status
+from CIME import utils
 from CIME.SystemTests.system_tests_common import SystemTestsCommon
 from CIME.case.case_setup import case_setup
 from CIME.XML.machines import Machines
-
+from CIME.config import ConfigBase
+from CIME.SystemTests import test_mods
 
 import evv4esm  # pylint: disable=import-error
 from evv4esm.__main__ import main as evv  # pylint: disable=import-error
 
-evv_lib_dir = os.path.abspath(os.path.dirname(evv4esm.__file__))
+version = evv4esm.__version_info__
+
+assert version[0] <= 0 and version[1] <= 5, "Please install evv4esm less than 0.5.x"
+
+EVV_LIB_DIR = os.path.abspath(os.path.dirname(evv4esm.__file__))
+
 logger = logging.getLogger(__name__)
-NINST = 30
+
+
+class MVKConfig(ConfigBase):
+    def __init__(self):
+        super().__init__()
+
+        if self.loaded:
+            return
+
+        self._set_attribute("component", "", "The main component.")
+        self._set_attribute(
+            "components", [], "Components that require namelist customization."
+        )
+        self._set_attribute("ninst", 30, "The number of instances.")
+        self._set_attribute(
+            "critical", 13, "The critical value for rejecting the null hypothese."
+        )
+        self._set_attribute(
+            "var_set", "default", "Name of the variable set to analyze."
+        )
+        self._set_attribute("ref_case", "Baseline", "Name of the reference case.")
+        self._set_attribute("test_case", "Test", "Name of the test case.")
+
+    def write_inst_nml(
+        self, case, set_nml_variable, component, iinst
+    ):  # pylint: disable=unused-argument
+        """Write per instance namelist.
+
+        This method is called once per instance to generate the namelist.
+
+        Args:
+            case (CIME.case.case.Case): The case instance.
+            write_nml_variable (function): Function takes two `str` arguments.
+            component (str): Component the namelist belongs to.
+            iinst (int): Instance unique number.
+        """
+        set_nml_variable("new_random", ".true.")
+        set_nml_variable("pertlim", "1.0e-10")
+        set_nml_variable("seed_custom", f"{iinst}")
+        set_nml_variable("seed_clock", ".true.")
+
+    def test_config(
```

Comment:
Could this instead be called `test_config_evv`? That way it's clear that this method is for creating the configuration to be passed to evv4esm?